mesos_slave/tests/test_e2e.py

```diff
@@ -0,0 +1,35 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import platform
+
+import pytest
+from six import iteritems
+
+from datadog_checks.mesos_slave import MesosSlave
+
+from .common import CHECK_NAME
+
+
+@pytest.mark.e2e
+# Linux only: https://github.com/docker/for-mac/issues/1031
+@pytest.mark.skipif(platform.system() != 'Linux', reason="Only runs on Unix systems")
+def test_e2e(dd_agent_check, instance):
+    aggregator = dd_agent_check(instance, rate=True)
+    check = MesosSlave(CHECK_NAME, {}, [{}])
+    metrics = {}
+    for d in (
+        check.SLAVE_TASKS_METRICS,
+        check.SYSTEM_METRICS,
+        check.SLAVE_RESOURCE_METRICS,
+        check.SLAVE_EXECUTORS_METRICS,
+        check.STATS_METRICS,
+    ):
+        metrics.update(d)
+
+    for _, v in iteritems(metrics):
+        aggregator.assert_metric(v[0])
+
+    aggregator.assert_all_metrics_covered()
+
+    aggregator.assert_service_check('mesos_slave.can_connect', status=check.OK)
```

Comment:
Could you factor the assertion logic ? e.g. https://github.com/DataDog/integrations-core/commit/bb4193689bb015e6855ef252293729f0e23ab506

Less code to maintain :)

The count is different for `aggregator.assert_service_check('mesos_slave.can_connect', count=1, status=check.OK)` 

One way is to use `count=2`, and call `check.check(instance)` in the integration test.